pkg/cli/debug/debug.go

```diff
@@ -594,6 +594,30 @@ func (o *DebugOptions) getContainerImageCommand(pod *corev1.Pod, container *core
 func (o *DebugOptions) transformPodForDebug(annotations map[string]string) (*corev1.Pod, []string) {
 	pod := o.Attach.Pod
 
+	if len(o.ToNamespace) > 0 {
+		svcAcctOrig := pod.Spec.ServiceAccountName
+		volumesOrig := pod.Spec.Volumes
+		clearVolumes(pod)
+		if len(pod.Spec.ServiceAccountName) != 0 {
+	        pod.Spec.ServiceAccountName = ""
+		}
+		if len(pod.Spec.DeprecatedServiceAccount) != 0 {
+	        pod.Spec.DeprecatedServiceAccount = ""
+		}
+		if len(pod.Spec.InitContainers) != 0 {
+			klog.V(4).Infof("PodSpec InitContainers cleared to run with '--to-namespace', run without '--to-namespace' to keep InitContainers")
+			pod.Spec.InitContainers = nil
+		}
+		if svcAcctOrig != pod.Spec.ServiceAccountName {
+			klog.V(4).Infof("PodSpec ServiceAccountName was cleared to run with '--to-namespace', run without '--to-namespace' to keep ServiceAccountName")
+		}
+		if len(pod.Spec.Volumes) != len(volumesOrig) {
+			if volumesOrig[0] != pod.Spec.Volumes[0] {
+				klog.V(4).Infof("PodSpec Volumes and VolumeMounts were cleared to run with '--to-namespace', run without '--to-namespace' to keep Volumes,VolumeMounts")
+			}
+		}
+	}
+
```

Comment:
@soltysh 
These all cause debug to fail  when run in a different namespace.
oc debug node: no change
oc debug everything-else: at least 1 of initContainers/Volumes/Mounts/ServiceAcct/ServiceAcctName cleared

Reply:
initContainers need to be cleared (almost always) because their commands almost always fail if not in original namespace - the initContainer cmd usually depends on VolumeMounts.  However, when debugging initContainers themselves, with `oc debug pod/name -c initContainerName` we will not clear the initContainers, bc in that case the cmd will be simply `/bin/sh`